### DISCUSSION TITLE
custom-style is only partially implemented for Opendocument / ODT?

### DISCUSSION THREAD

**Discussion by iandol** [jgm/pandoc#9656] [General]
Based on the manual and this open issue (#2106) I surmised the use of a `custom-style` was not working for ODT output. However, I found this message:

https://github.com/jgm/pandoc/commit/bde3d7622ddd2cda7adabc309335bb2865c1d520

This applies to `div` [block]elements. but currently there is no `span` [inline] support:

```
pandoc -t opendocument
[Get out]{custom-style="Strikeout"} he said.

::: {custom-style="Comment"}
This is [text with a character style]{custom-style="Teletype"}. And this is also [text with a character style]{custom-style="Highlighted"} text style.
:::

<text:p text:style-name="Text_20_body">Get out he said.</text:p>
<text:p text:style-name="Comment">This is text with a character style.
And this is also text with a character style text style.</text:p>
```

Did the addition of support for block elements (the code change is fairly minimal, though I don't undrstand Haskell) make it easier to implement inline ones?

**Comment by iandol** [comment]
For character styles the opendocument output should look like:

```xml
<text:span text:style-name="Strikeout">Get out</text:span>
```

It appears fairly easy to implement?

**Reply by jgm** [reply]
The relevant change is actually https://github.com/jgm/pandoc/commit/a2ce28fa4d0e90fc2cd1e4d3a6d738005323164f

**Reply by jgm** [reply]
I don't think it would be too hard to add support for inline custom-styles.
We'd need to modify the TextStyle to have a constructor like Custom Text.

**Reply by jgm** [reply]
Go ahead and put up an issue for this if you'd like.

**Reply by iandol** [reply]
Thanks as always Jon, this is now issue #9657

**Reply by iandol** [reply]
Just to update this discussion: Jon fixed this in https://github.com/jgm/pandoc/commit/bc9f9902566f6e2cdd100d16f82deaacd18acd79 (I confirmed it is working in the last nightly) — we now have both block and span custom styles working in OpenDocument / ODT. A big thank you to @jgm!
